### PR TITLE
Ventcrawling delay

### PR DESCRIPTION
**Pull request by DamianX**
Added a 1 second delay to crawling out of a vent and added a visible message indicating that.
:cl:
 * tweak: Coming out of a vent/scrubber while ventcrawling now has a 1 second delay and associated warning message.

### CODE REVIEW

**Comment by ghost** [comment]
Does the warning message show to people outside the vent?

**Comment by DamianX** [comment]
Yes, that's the whole point.

**Comment by ghost** [comment]
>nerfs vent ganking

**Comment by yclatious** [comment]
Nerfing vent ganking just seems wrong to me.

**Comment by PJB3005** [comment]
Make the vent visually shake and THEN we're talking.

**Comment by DamianX** [comment]
k

**Review by PJB3005 — APPROVED** [review]
My move is hitting approve!

**Comment by ihadtoregisterforthis** [comment]
For comparison, what's the lenght of the entry delay?
Can it be cancelled? 
How much does the vent shake?
For mobs that get pressure/heat damage from pipes, or borers, a second can be a great deal.

**Comment by DamianX** [comment]
Seems to be 4.5s to enter vents
It can be cancelled the same way other progress-bar actions can be cancelled
The vent shake duration is double the duration of a kick to any machinery, and the shake intensity is the same as a kick.

**Comment by clusterfack** [comment]
This would also stop the issue of mommies accidentally flying out of vents.

**Comment by Fruktlimpa** [comment]
What sparked the idea for this PR? Xeno nerf? MoMMI QoL? 
If you haven't welded vents at the first sign of Xeno then you deserve to get ganked.

**Comment by PJB3005** [comment]
@clusterfack while I agree, there's better ways to fix that.

Hint: remapping the pipe network to not be dumb.

**Comment by DamianX** [comment]
@Fruktlimpa the reason is someone on discord vaguely mentioned it

**Comment by clusterfack** [comment]
I mentioned it on discord for a number of reasons

1. Moving to the nonpipe teleporting system of ventcrawling actually massively advantaged things that can ventcrawl that have instant stuns. Examples include mommis with emag arm and xenos with instant disarm stun.

In that respect I dont think the vent should shake, because that makes it TOO obvious it is occuring, instead of just alerting people. I think I'd prefer a sound over a visible shake on the particular vent.

2. This also happens to help accidentally jumping out of vents. Which in a lot of ways helps things that are ventcrawling because fucking up and running out of a vent at the wrong time can instakill you as Xeno.

**Comment by Fruktlimpa** [comment]
I agree that ventcrawling with a capable stun is probably one of the strongest things in the game and I really don't think most people will react in time even when it is delayed by one second.
I just feel that it is a nerf to something that so very rarely happens. Emagged MoMMI's aren't common.

But I don't really think one second will be that big of a deal so I'm fine with it.
Since it also fixes the issue of accidentally exiting vents, which has happened all to often, it is fine.

File: code/ATMOSPHERICS/atmospherics.dm [hunk 1]
```diff
@@ -349,9 +349,12 @@ Pipelines + Other Objects -> Pipe network
...
 		if(is_type_in_list(target_move, ventcrawl_machinery) && target_move.can_crawl_through())
-			user.remove_ventcrawl()
-			user.forceMove(target_move.loc) //handles entering and so on
-			user.visible_message("You hear something squeezing through the ducts.", "You climb out the ventilation system.")
+			user.visible_message("Something is squeezing through the ducts...", "You start crawling out the ventilation system.")
+			target_move.shake(2, 3)
+			if(do_after(user, target_move, 10))
+				user.remove_ventcrawl()
+				user.forceMove(target_move.loc) //handles entering and so on
+				user.visible_message("You hear something squeezing through the ducts.", "You climb out the ventilation system.")
```
Comment (Intigracy) [inline code/ATMOSPHERICS/atmospherics.dm]: This should probably be "You hear something squeeze through the ducts."

**Pull request merged**